paymentintent_test.go

```diff
@@ -0,0 +1,53 @@
+package stripe
+
+import (
+	"encoding/json"
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+)
+
+func TestPaymentIntent_UnmarshalJSON(t *testing.T) {
+	intentData := map[string]interface{}{
+		"id":     "pi_123",
+		"object": "payment_intent",
+		"allowed_source_types": []interface{}{
+			"card",
+		},
+		"charges": map[string]interface{}{
+			"object":   "list",
+			"has_more": true,
+			"data": []map[string]interface{}{
+				{
+					"id":     "ch_123",
+					"object": "charge",
+				},
+				{
+					"id":     "ch_234",
+					"object": "charge",
+				},
+			},
+		},
+		"next_source_action": map[string]interface{}{
+			"type": "authorize_with_url",
+			"value": map[string]interface{}{
+				"url": "https://stripe.com",
+			},
+		},
+	}
+
+	bytes, err := json.Marshal(&intentData)
+	assert.NoError(t, err)
+
+	var intent PaymentIntent
+	err = json.Unmarshal(bytes, &intent)
+	assert.NoError(t, err)
+
+	assert.Equal(t, "pi_123", intent.ID)
+
+	assert.Equal(t, 1, len(intent.AllowedSourceTypes))
+	assert.Equal(t, 2, len(intent.Charges.Data))
+
+	assert.Equal(t, PaymentIntentNextActionAuthorizeWithURL, intent.NextSourceAction.Type)
+	assert.Equal(t, "https://stripe.com", intent.NextSourceAction.Value.AuthorizeWithUrl.URL)
+}
```

Comment:
Would you mind adding a test specifically for `PaymentIntentSourceAction`'s `UnmarshalJSON`? IMO, we should try to have a specific test for every type that has a custom `UnmarshalJSON` implementation.

It's fairly similar to `ExternalAccount`'s unmarshaling, so I'd use the test `TestExternalAccount_UnmarshalJSON` in `account_test.go` for inspiration.

We could also then remove the `next_source_action` from `TestPaymentIntent_UnmarshalJSON` now that there's a more local test to check the same thing, but it's up to you.